config/300-clustertask.yaml

```diff
@@ -24,23 +24,28 @@ metadata:
 spec:
   group: tekton.dev
   preserveUnknownFields: false
-  validation:
-    openAPIV3Schema:
-      type: object
-      # One can use x-kubernetes-preserve-unknown-fields: true
-      # at the root of the schema (and inside any properties, additionalProperties)
-      # to get the traditional CRD behaviour that nothing is pruned, despite
-      # setting spec.preserveUnknownProperties: false.
-      #
-      # See https://kubernetes.io/blog/2019/06/20/crd-structural-schema/
-      # See issue: https://github.com/knative/serving/issues/912
-      x-kubernetes-preserve-unknown-fields: true
   versions:
-  - name: v1alpha1
+  - &version
+    name: v1alpha1
     served: true
     storage: false
-  - name: v1beta1
-    served: true
+    schema:
+      openAPIV3Schema:
+        type: object
+        # One can use x-kubernetes-preserve-unknown-fields: true
+        # at the root of the schema (and inside any properties, additionalProperties)
+        # to get the traditional CRD behaviour that nothing is pruned, despite
+        # setting spec.preserveUnknownProperties: false.
+        #
+        # See https://kubernetes.io/blog/2019/06/20/crd-structural-schema/
+        # See issue: https://github.com/knative/serving/issues/912
+        x-kubernetes-preserve-unknown-fields: true
+    # Opt into the status subresource so metadata.generation
+    # starts to increment
+    subresources:
+      status: {}
+  - <<: *version
```

Comment:
Given that *version is only used once in this file (and each file contains its own *version definition), is there a reason to use this syntax as opposed to just replacing `&version` above with
```
name: v1beta1
storage: true
```
?

Reply:
if we don't use that notation, we need to repeat all the field, so `storage`, `served` and `schema` :upside_down_face:
As is, it duplicates the definition from `v1alpha1` and only override the the `name` and `storage` field.

Reply:
Ahh, ok...I did not fully understand how << worked. Makes sense now.